### PR TITLE
[SPARK-19598][SQL]Remove the alias parameter in UnresolvedRelation

### PR DESCRIPTION
**Pull request by windpiger**
## What changes were proposed in this pull request?

Remove the alias parameter in `UnresolvedRelation`, and use `SubqueryAlias` to replace it.
This can simplify some `match case` situations.

For example, the broadcast hint pull request can have one fewer case https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveHints.scala#L57-L61

## How was this patch tested?
add some unit tests

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveHints.scala [hunk 1]
```diff
@@ -54,10 +54,6 @@ object ResolveHints {
 
       val newNode = CurrentOrigin.withOrigin(plan.origin) {
         plan match {
-          case r: UnresolvedRelation =>
```
Comment (rxin) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveHints.scala]: actually you can't remove this case entirely. you still need to match on UnresolvedRelation, but just use `r.tableIdentifier.table`

Comment (windpiger) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveHints.scala]: oh, yes, remove the case will miss  the hint on table without alias

**Comment by gatorsmile** [comment]
I like this change. 

To run unit test cases in your local environment, below are some common commands when I develop Spark SQL

- To do the local style check
dev/lint-scala

- To run the SQL test cases in your local laptop:
build/sbt -Phive hive/test
build/sbt sql/test
build/sbt catalyst/test

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala [hunk 2]
```diff
@@ -645,17 +645,21 @@ class AstBuilder extends SqlBaseBaseVisitor[AnyRef] with Logging {
...
   }
 
   /**
    * Create an aliased table reference. This is typically used in FROM clauses.
    */
   override def visitTableName(ctx: TableNameContext): LogicalPlan = withOrigin(ctx) {
-    val table = UnresolvedRelation(
-      visitTableIdentifier(ctx.tableIdentifier),
-      Option(ctx.strictIdentifier).map(_.getText))
-    table.optionalMap(ctx.sample)(withSample)
```
Comment (gatorsmile) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala]: This should be kept in the last line

**Comment by windpiger** [comment]
@gatorsmile thanks for your good suggestion～Usually I run the single test case in IDEA，and let jenkins run testcases globaly，I will change the bad behavior， thanks a lot!

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala (hunk 2 above)
Comment (windpiger) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala]: yes, I can not modify the original default expression order between alias and and sample.

**Comment by windpiger** [comment]
retest this please

**Comment by windpiger** [comment]
@rxin @gatorsmile help to continue to review this? Thanks :)

**Comment by gatorsmile** [comment]
LGTM

**Comment by gatorsmile** [comment]
retest this please

**Comment by gatorsmile** [comment]
Thanks! Merging to master.

**Pull request closed without merging**